src/test/librbd/journal/test_Entries.cc

```diff
@@ -196,6 +196,70 @@ TEST_F(TestJournalEntries, AioDiscard) {
   ASSERT_EQ(234U, aio_discard_event.length);
 }
 
+TEST_F(TestJournalEntries, AioDiscardWithPrune) {
+  REQUIRE_FEATURE(RBD_FEATURE_JOURNALING);
+
+  // The discard path can create multiple image extents (ImageRequest.cc) in the
+  // case where the discard request needs to be pruned and multiple objects are
+  // involved in the request. This test ensures that journal event entries are
+  // queued up for each image extent.
+
+  // Create an image that is multiple objects so that we can force multiple
+  // image extents on the discard path.
+  CephContext* cct = reinterpret_cast<CephContext*>(_rados.cct());
+  auto size = 4 * (1ull << cct->_conf.get_val<uint64_t>("rbd_default_order"));
+
+  auto image_name = get_temp_image_name();
+  ASSERT_EQ(0, create_image_pp(m_rbd, m_ioctx, image_name, size));
+
+  librbd::ImageCtx *ictx;
+  ASSERT_EQ(0, open_image(image_name, &ictx));
+
+  const int64_t object_size = std::min<int64_t>(ictx->size, 1 << ictx->order);
```

Comment:
Ah, what happened here is that originally I tried using the volume that's created by the fixture, but then I later realized that it was too small for my needs and created my own. I've cleaned up the logic around object/volume size in both tests so that it's less confusing (and confirmed that the stress test still reproduces the reported issue).